File: z-clients/javascript/src/models/msgStreamCommitIn.ts
// this file is @generated

export interface MsgStreamCommitIn {
    namespace?: string | null;
    offset: number;
}

export interface MsgStreamCommitIn_ {
    namespace?: string | null;
    topic: string;
    consumerGroup: string;
    offset: number;
}

export const MsgStreamCommitInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgStreamCommitIn_ {
        return {
            namespace: object['namespace'],
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            offset: object['offset'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgStreamCommitIn_): any {
        return {
            'namespace': self.namespace,
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'offset': self.offset,
        };
    }
}

File: z-clients/python/coyote/models/msg_stream_commit_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgStreamCommitIn(BaseModel):
    namespace: t.Optional[str] = None

    offset: int


class _MsgStreamCommitIn(BaseModel):
    namespace: t.Optional[str] = None

    topic: str

    consumer_group: str

    offset: int
